packages/gensx-client/src/sdk.ts

```diff
@@ -0,0 +1,601 @@
+  format?: "sse" | "ndjson" | "json";
+  // Allow overriding client-level config
+  org?: string;
+  project?: string;
+  environment?: string;
+}
+
+export interface StartOptions {
+  inputs?: Record<string, unknown>;
+  // Allow overriding client-level config
+  org?: string;
+  project?: string;
+  environment?: string;
+}
+
+export interface StartResponse {
+  executionId: string;
+  executionStatus: string;
+  data?: unknown;
+}
+
+export interface GetProgressOptions {
+  executionId: string;
+  format?: "sse" | "ndjson";
+}
+
+export interface WorkflowExecution {
+  executionId: string;
+  status: string;
+  progress?: unknown[];
+  result?: unknown;
+  error?: unknown;
+}
+
+type RunReturn<
+  TOutput,
+  TStream extends boolean | undefined = undefined,
+> = TStream extends true
+  ? {
+      outputStream: AsyncIterable<TOutput>;
+      progressStream: ReadableStream;
+    }
+  : {
+      output: TOutput;
+      progressStream: ReadableStream;
+    };
+
+/**
+ * GenSX SDK for interacting with GenSX workflows
+ *
+ * Usage:
+ * ```typescript
+ * const gensx = new GenSX({
+ *   apiKey: 'your-api-key',
+ *   org: 'your-org',
+ *   project: 'your-project',
+ *   environment: 'production'
+ * });
+ *
+ * // Run a workflow with just inputs
+ * const { output } = await gensx.run<string>('workflowName', {
+ *   inputs: { userMessage: 'Hello world' }
+ * });
+ *
+ * // Or override org/project for a specific call
+ * const { output } = await gensx.run<string>('workflowName', {
+ *   inputs: { userMessage: 'Hello world' },
+ *   org: 'different-org',
+ *   project: 'different-project'
+ * });
+ * ```
+ */
+export class GenSX {
+  private apiKey?: string;
+  private baseUrl: string;
+  private org?: string;
+  private project?: string;
+  private environment?: string;
+  private isLocal: boolean;
+
+  constructor(config: GenSXConfig) {
+    this.baseUrl = config.baseUrl ?? "https://api.gensx.com";
+    this.isLocal =
+      this.baseUrl.includes("localhost") && !config.overrideLocalMode;
+
+    if (!this.isLocal) {
+      // For non-local mode, require apiKey
+      this.apiKey = this.getApiKey(config.apiKey);
+      if (!this.apiKey) {
+        throw new Error(
+          "apiKey is required. Provide it in the constructor or set the GENSX_API_KEY environment variable.",
+        );
+      }
+
+      // For non-local mode, require org/project/environment
+      if (!config.org || !config.project || !config.environment) {
+        throw new Error(
+          "org, project, and environment are required when calling deployed workflows. Provide them in the constructor or method options.",
+        );
+      }
+    }
+
+    this.org = config.org;
+    this.project = config.project;
+    this.environment = config.environment;
+  }
+
+  /**
+   * Run a workflow with optional streaming
+   * @returns Either { output, progressStream } or { outputStream, progressStream } based on stream flag
+   */
+  async run<TOutput = unknown>(
```

Comment:
run should be generic over <TInput, TOutput>, no? 

Then RuntimeOptions should also be generic over the input type. `RuntimeOptions<TInput>`